test/ResultTest.cpp

```diff
@@ -400,6 +413,67 @@ TEST(Result, verifyApplyLimitOffsetHandlesZeroLimitCorrectly) {
   }
 }
 
+// _____________________________________________________________________________
+TEST(Result, verifyApplyLimitOffsetHandlesNonZeroOffsetWithoutLimitCorrectly) {
+  auto idTable = makeIdTableFromVector({{0, 7}, {1, 6}, {2, 5}, {3, 4}});
+  LimitOffsetClause limitOffset{std::nullopt, 1};
+  {
+    uint32_t callCounter = 0;
+    Result result{idTable.clone(), {}, LocalVocab{}};
+    result.applyLimitOffset(
+        limitOffset, [&](std::chrono::microseconds, const IdTable& innerTable) {
+          EXPECT_EQ(innerTable.numRows(), 3);
+          ++callCounter;
+        });
+    EXPECT_EQ(callCounter, 1);
+  }
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    uint32_t callCounter = 0;
+    Result result{std::move(generator), {}, LocalVocab{}};
+    result.applyLimitOffset(
+        limitOffset, [&](std::chrono::microseconds, const IdTable& innerTable) {
+          for (const auto& row : innerTable) {
+            ASSERT_EQ(row.size(), 2);
+            // Make sure we never get values that were supposed to be filtered
+            // out.
+            EXPECT_NE(row[0].getVocabIndex().get(), 0);
+            EXPECT_NE(row[1].getVocabIndex().get(), 7);
+          }
+          ++callCounter;
+        });
+
+    consumeGenerator(result.idTables());
+
+    EXPECT_GE(callCounter, 1);
+  }
```

Comment:
You now have at least three very redundant tests with a rather sofisticated logic.

1. Factor out the common code to a helper (you basically have an input table, a LimitOffsetClause and an expected result), document and reuse that helper.

2. I think the easier way to test (and even more precise) is to just concatenate all the output tables (via insertAtEnd or something, or even a helper function that we already have that does that, and then check for equality.